feat: added new SignMessage for signinf simple string message and message to flag signed typed data are not support

GaetanoMondelli opened the pull request.

Commit:
feat: added new SignMessage for signinf simple string message and flag for not supported signedTypedData

traylewin reviewed (commented):
Please update Chain members as requested in updated Chain interface
Please check all comments are correct
Approved after making these changes

traylewin commented on src/plugin/algoSignMessage.ts:
```diff
@@ -0,0 +1,137 @@
+  private _data: any
+
+  private applyOptions(options: Models.SignMessageOptions) {
+    this._options = options ? options : { signMethod: 'algorand-sign'}
```
This value should be 'default' so that it is the same across all chains:
e.g. SignMethod.Default - 'default'
Use an Enum and export the type like other types

traylewin commented on src/plugin/models/cryptoModels.ts:
```diff
@@ -27,3 +27,8 @@ export type AlgoEncryptionOptions = Crypto.Ed25519Crypto.Ed25519PasswordEncrypti
 export type EncryptionOptions = AlgoEncryptionOptions
+
+/** An object containing the string message to sign */
+export type AlgoSignDataInput = {
```
Please move to generalModels

Commit:
fix: PR comments
- renamed supportsTypedDataSignature to supportsSignMessageTypedData and added more specific comment
- data to Message in fields and functions in  eosSignMessage
- moved AlgoSignDataInput to general models

Commit:
fix: message/data in signMessage and AlgoSignMessage from any to string

traylewin commented on src/plugin/models/generalModels.ts:
```diff
@@ -36,3 +36,13 @@ export enum AlgorandUnit {
 export type AlgorandSymbol = string & Models.ChainSymbolBrand
+
+/** SignMethod enum contains the methods that can be used to sign a message */
+export enum SignMethod {
```
since SignMethod and AlgoSignDataInput are both only used for signMessage, this should be clear here. please rename:
SignMethod => AlgorandSignMessageMethod
AlgoSignDataInput => AlgorandSignMessageDataInput 
note: please dont shorten the prefix from Algorand - as 'Algo' is the currency name and can add confusion

GaetanoMondelli replied:
Done!
Noted, I got confused with the file naming convention, will keep this in mind.

traylewin reviewed (commented):
Approved after minor changes - feel free to merge to master

traylewin reviewed (approved):
Approved after minor changes are made - feel free to merge to master after

Commit:
fix: rename of algorand classes
